app/assets/javascripts/tylium/modules/auto_save/local.js

```diff
@@ -0,0 +1,144 @@
+class LocalAutoSave {
+  constructor(target) {
+    if (target.tagName !== 'FORM') { console.error('Can\'t initialize local auto save on anything but a form'); return; }
+    this.target = target;
+    this.key = target.dataset.autoSaveKey;
+    this.cancelled = false;
+
+    // List of available inputs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+    // Only permit these inputs to be saved so that it does not store unnecessary data in local cache
+    this.permittedInputTypes = [
+      'checkbox',
+      'color',
+      'date',
+      'email',
+      'hidden', // Needed for hidden tag_input in issues/form
+      'number',
+      'radio',
+      'tel',
+      'text'
+    ];
+
+    // Don't store authenticity_token and utf8
+    this.excludedInputNames = ['utf8', 'authenticity_token', '_method'];
+
+    this.init();
+  }
+
+  init() {
+    this.behaviors();
+    this.restoreData();
+  }
+
+  behaviors() {
+    var that = this;
+
+    this.target.addEventListener('submit', function(event) {
+      localStorage.removeItem(that.key);
+    });
+
+    var clearCacheElement = this.target.querySelector('[data-behavior~=clear-local-auto-save]');
+
+    if (clearCacheElement) {
+      clearCacheElement.addEventListener('click', function(event) {
+        that.cancelled = true;
+        localStorage.removeItem(that.key);
+      })
+    }
+
+    // Find all inputs in the form, then exclude based on excluded input types
+    var formInputs = Array.from(this.target.querySelectorAll('input')).filter(function(input) {
+      return that.permittedInputTypes.includes(input.getAttribute('type')) && !that.excludedInputNames.includes(input.name);
+    })
+
+    formInputs = formInputs.concat(Array.from(this.target.querySelectorAll('textarea, select')));
```

Comment:
It's because of the permittedInputTypes in line 10. Select and textarea does not have a type attribute. Now that you have mentioned it, I've combined it with this
```
var formInputs = Array.from(this.target.querySelectorAll('input, textarea, select')).filter(function(input) {
  return (['SELECT', 'TEXTAREA'].includes(input.tagName) ||
              (that.permittedInputTypes.includes(input.getAttribute('type')) && !that.excludedInputNames.includes(input.name)));
})
```

Reply:
Ignore this, i've removed the formInputs completeley.